File: src/main/java/io/koju/autopos/catalog/domain/Category.java
package io.koju.autopos.catalog.domain;

import io.koju.autopos.kernel.domain.AuditableBaseEntity;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import static javax.persistence.GenerationType.SEQUENCE;

@Entity
@Table(name = "category")
@Getter
@Setter
public class Category extends AuditableBaseEntity {

    @Id
    @SequenceGenerator(name = "category_id_seq", sequenceName = "category_id_seq", allocationSize = 1)
    @GeneratedValue(strategy = SEQUENCE, generator = "category_id_seq")
    private Long id;


    @NotNull
    @Size(min = 2, max = 3)
    @Column(name = "short_name", length = 3, nullable = false)
    private String shortName;

    @NotNull
    @Size(min = 3, max = 50)
    @Column(name = "name", length = 50, nullable = false)
    private String name;

}

File: src/main/scala/io/koju/autopos/catalog/domain/Category.scala
package io.koju.autopos.catalog.domain

import java.lang.Long
import javax.persistence.GenerationType.SEQUENCE
import javax.persistence._
import javax.validation.constraints.{NotNull, Size}

import io.koju.autopos.kernel.domain.AuditableBaseEntity

import scala.beans.BeanProperty

@Entity
@Table(name = "category")
class Category extends AuditableBaseEntity {

  @Id
  @SequenceGenerator(name = "category_id_seq", sequenceName = "category_id_seq", allocationSize = 1)
  @GeneratedValue(strategy = SEQUENCE, generator = "category_id_seq")
  @BeanProperty
  var id: Long = _

  @NotNull
  @Size(min = 2, max = 3)
  @Column(name = "short_name", length = 3, nullable = false)
  @BeanProperty
  var shortName: String = _

  @NotNull
  @Size(min = 3, max = 50)
  @Column(name = "name", length = 50, nullable = false)
  @BeanProperty
  var name: String = _

}
